### PR TITLE
Fetches the manifest for Tekton Dashboard through init Controller

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
Signed-off-by: Puneet Punamiya <ppunamiy@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

### CODE REVIEW

File: pkg/reconciler/kubernetes/initcontroller/initcontroller.go [hunk 1]
```diff
@@ -32,9 +32,10 @@ import (
 )
 
 type Controller struct {
-	Manifest         *mf.Manifest
-	Logger           *zap.SugaredLogger
-	VersionConfigMap string
+	Manifest                *mf.Manifest
+	Logger                  *zap.SugaredLogger
+	VersionConfigMap        string
+	IsDashboardSpecReadOnly bool
```
Comment (nikhil-thomas) [inline pkg/reconciler/kubernetes/initcontroller/initcontroller.go:38]: this looks a bit hacky and out of place.

let us create a an `PayloadOptions` struct and for now let us add 1 property `ReadOnly`

```bash
type PayloadOptions struct {

    ReadOnly bool
}
```

Then in the `func (ctrl Controller) fetchSourceManifests(ctx context.Context) error`
can become `func (ctrl Controller) fetchSourceManifests(ctx context.Context, opt PayloadOptions) error`

and the 2 calls could become,

`fullaccessManifest, _ := ctrl.InitController(ctx, PayloadOptions{ReadOnly: false})`

and 

`readOnlyManifest, _ := ctrl.InitController(ctx, PayloadOptions{ReadOnly: true})`

Comment (PuneetPunamiya) [reply pkg/reconciler/kubernetes/initcontroller/initcontroller.go:38]: What would be the readOnly value for tektonPipeline and tektonTriggers ?

Comment (nikhil-thomas) [reply pkg/reconciler/kubernetes/initcontroller/initcontroller.go:38]: for other reconcilers, we will not need `readOnlyManifest, _ := ctrl.InitController(ctx, PayloadOptions{ReadOnly: true})` 

there will be only one call, for full access.

Comment (nikhil-thomas) [reply pkg/reconciler/kubernetes/initcontroller/initcontroller.go:38]: ie, 
`fullaccessManifest, _ := ctrl.InitController(ctx, PayloadOptions{})`

Comment (PuneetPunamiya) [reply pkg/reconciler/kubernetes/initcontroller/initcontroller.go:38]: got it thanks will update the pr 👍🏻

**Pull request merged**